doc/algorithms/index.rst

```diff
@@ -0,0 +1,20 @@
+######################
+First-Party Algorithms
+######################
+
+Amazon SageMaker provides implementations of some common machine learning algorithms optimized for GPU architecture and massive datasets.
+
+.. toctree::
+    :maxdepth: 2
+
+    sagemaker.amazon.amazon_estimator
+    factorization_machines
+    ipinsights
+    kmeans
+    knn
+    lda
+    linear_learner
+    ntm
+    object2vec
+    pca
+    randomcutforest
```

Comment:
doesn't have to be in this PR, but is there a way to automate getting the list of algorithms (maybe using the glob trick)?

Reply:
yes, we can totally do that ... it'll be alphabetical, so if we want any organization, then we'd want to rename the files to fit that org.